src/type_checkers/variable.cr

```diff
@@ -57,7 +57,12 @@ module Mint
           when Ast::Statement, Ast::WhereStatement
             case item.target
             when Ast::TupleDestructuring
-              type.parameters[value[1]]
+              case val = value[1]
+              when Int32
+                type.parameters[val]
+              when Array(Int32)
+                val.reduce(type) { |curr_type, curr_val| curr_type.parameters[curr_val] }
+              end.not_nil!
```

Comment:
```suggestion
              in Int32
                type.parameters[val]
              in Array(Int32)
                val.reduce(type) { |curr_type, curr_val| curr_type.parameters[curr_val] }
              end
```

Reply:
Not sure how I've gone this long without knowing about this!

Reply:
Me neither :joy: This needs be documented here https://tw.crystal-lang.org/docs/syntax_and_semantics/case.html :D